engine/swarm/secrets.md

```diff
@@ -0,0 +1,770 @@
+
+    ```bash
+    $ docker service  create --name="redis" --secret="my_secret_data" redis:alpine
+    ```
+
+3.  Verify that the task is running without issues using `docker service ps`. If
+    everything is working, the output looks similar to this:
+
+    ```bash
+    $ docker service ps --no-trunc redis
+
+    NAME                               lIMAGE         NODE              DESIRED STATE  CURRENT STATE           ERROR  PORTS
+    redis.1.cezsyv6xljjj6bzpajbc6agkb  redis:alpine   ip-172-31-46-109  Running        Running 26 seconds ago
+    ```
+
+    If there were an error, and the task were failing and repeatedly restarting,
+    you would see something like this:
+
+    ```bash
+    $ docker service ps redis
+
+    NAME                      IMAGE         NODE  DESIRED STATE  CURRENT STATE          ERROR                      PORTS
+    redis.1.siftice35gla      redis:alpine  moby  Running        Running 4 seconds ago                             
+     \_ redis.1.whum5b7gu13e  redis:alpine  moby  Shutdown       Failed 20 seconds ago      "task: non-zero exit (1)"  
+     \_ redis.1.2s6yorvd9zow  redis:alpine  moby  Shutdown       Failed 56 seconds ago      "task: non-zero exit (1)"  
+     \_ redis.1.ulfzrcyaf6pg  redis:alpine  moby  Shutdown       Failed about a minute ago  "task: non-zero exit (1)"  
+     \_ redis.1.wrny5v4xyps6  redis:alpine  moby  Shutdown       Failed 2 minutes ago       "task: non-zero exit (1)"
+    ```
+
+4.  Get the ID of the `redis` service task container using `docker ps` , so that
+    you can use `docker exec` to connect to the container and read the contents
+    of the secret data file, which defaults to being readable by all and has the
+    same name as the name of the secret.  The ID is the first field of the
+    output of `docker ps`.
+
+    ```bash
+    $ docker ps --filter name=redis
+
+    CONTAINER ID        IMAGE                                                                           COMMAND                  CREATED             STATUS              PORTS               NAMES
+    b9fc6c983b36        redis@sha256:b4c47064c182d77aed359ceab16ebfec9b749a6b18af3988731181252f0f1b44   "docker-entrypoint..."   2 hours ago         Up 2 hours          6379/tcp            redis.1.loctgnmbx02fbved667wzfdnq
+
+    $ docker exec b9fc6c983b36 ls -l /run/secrets
+
+    total 4
+    -r--r--r--    1 root     root            17 Dec 13 22:48 my_secret_data
+
+    $ docker exec b9fc6c983b36 cat /run/secrets/my_secret_data
+
+    This is a secret
+    ```
+
+5.  Verify that the secret is **not** available if you commit the container.
+
+    ```bash
+    $ docker commit b9fc6c983b36 committed_redis
+
+    sha256:dfde27e778c34d2c402d96bda227ecaa906b6bf1eb97ffd3bf00bb7485408153
+
+    $ docker run --rm -it committed_redis cat /run/secrets/my_secret_data
+
+    cat: can't open '/run/secrets/my_secret_data': No such file or directory
+    ```<!--'fix-->
```

Comment:
It's intentional to fix the unbalanced `'` in `can't`, otherwise the rest of the file is screwed up in Atom! I'll move it closer to the source of the problem.